packages/electron-commons/src/initializers/electron-node.ts

```diff
@@ -76,7 +82,17 @@ export const initializeNodeEnvironment: EnvironmentInitializer<
         }
     );
 
-    child.send({ id: 'nodeStartupOptions', runOptions: nodeEnvStartupArguments } as INodeEnvStartupMessage);
+    const envInitFailed = deferred();
```

Comment:
- deferred promises are a REALLY good way to make code harder to read. we should avoid if possible
- promise itself is named envInitFailed, which is odd. this is also why you've Promise.raced at the bottom. it never resolves (even when init failed). instead it only rejects. the double negatives are confusing.
- having a promise for envReady AND envInitFailed is confusing. I'd expect **envReady** to reject if anything fails. not to have a separate promise that only conditionally fails

Reply:
removed handling of env init failing, it would be hard to make this in same promise, cause envIsReady promise is created in communication, but communication know nothing about env init failure...
there is no init fail handling in current version, so if we do need this, it is better to implement separately